src/components/TextArea/TextArea.tsx

```diff
@@ -0,0 +1,142 @@
+/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
+import '../../yamui';
+import * as React from 'react';
+import { BaseComponentProps } from '../../util/BaseComponent/props';
+import { TextField as FabricTextField } from 'office-ui-fabric-react/lib/TextField';
+import { Async } from 'office-ui-fabric-react/lib/Utilities';
+import './TextArea.css';
+
+export interface TextAreaProps extends BaseComponentProps {
+  /**
+   * Current value of the TextArea.
+  */
+  value?: string;
+
+  /**
+   * Placeholder text. Displayed until a value is entered.
+  **/
+  placeHolder?: string;
+
+  /**
+   * Error message string.
+   */
+  errorMessage?: string;
+
+  /**
+   * Disabled state of the TextArea.
+  */
+  disabled?: boolean;
+
+  /**
+   * Label for the TextArea.
+  */
+  label?: string;
+
+  /**
+   * The TextArea input description
+   */
+  description?: string;
+
+  /**
+   * Use to indicate that a value must be provided to allow the containing form to be submitted.
+  */
+  required?: boolean;
+
+  /**
+   * Height of the field, in number of rows.
+  **/
+  rows?: number;
+
+  /**
+   * Whether or not to auto adjust textField height.
+  **/
+  autoAdjustHeight?: boolean;
+
+  /**
+   * Callback for the onChanged event.
+   */
+  onChange?: (newValue: any) => void;
+
+  /**
+   * TextArea will trigger `onChange` after users stop typing for `onChangeDebounceTime`
+   * milliseconds.
+   */
+  onChangeDebounceTime?: number;
+
+  /**
+   * Focus callback handler.
+   */
+  onFocus?: (event: React.FocusEvent<HTMLInputElement | HTMLTextAreaElement>) => void;
+
+  /**
+   * Blur callback handler.
+   */
+  onBlur?: (event: React.FocusEvent<HTMLInputElement | HTMLTextAreaElement>) => void;
+
+  /**
+   * Mouse enter callback handler.
+   */
+  onMouseEnter?: (event: React.MouseEvent<HTMLInputElement | HTMLTextAreaElement>) => void;
+
+  /**
+   * Mouse leave callback handler.
+   */
+  onMouseLeave?: (event: React.MouseEvent<HTMLInputElement | HTMLTextAreaElement>) => void;
+}
+
+/**
+ * The TextArea component enables a user to type text into an app. It's typically used to capture
+ * a single line of text, but can be configured to capture multiple lines of text. The text
+ * displays on the screen in a simple, uniform format.
+ */
+export default class TextArea extends React.PureComponent<TextAreaProps, {}> {
```

Comment:
I would have first extracted our Textfield implementation + styling to BaseTextField, then added the necessary TextArea functionality to BaseTextField

Reply:
My ruby experience has taught me that taking the path of independently implementing the two (Ideally 3) like things, and then extracting the common parts usually leads to better abstractions. I guess we'll see. :-)